File: go/patterns/twosum/main_test.go
package main

import (
    "errors"
    "testing"
)

func TestTwoSum(t *testing.T) {
    tests := []struct {
        name string
        nums []int
        target int
        wantErr error
        wantResp []int
    } {
        {
            name: "Correct Test",
            nums: []int{1, 2, 3, 4},
            target: 7,
            wantErr: nil,
            wantResp: []int{2, 3},
        },
    }
    for _, tt := range tests {
        t.Run(tt.name, func(t *testing.T) {
            gotResp, gotErr := TwoSum(tt.nums, tt.target)
        })
    }

}

File: rust/tmps/tmp3/hello/src/main_test.rs
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test() {
        let nums = vec![1, 2, 3, 4];
        let target = 7;
        let result = run(&nums, target);

        let expected = vec![2, 3];

        assert_eq!(result, expected);
    }
}
